src/test/java/org/opentripplanner/middleware/ApiKeyManagementTest.java

```diff
@@ -1,27 +1,167 @@
 package org.opentripplanner.middleware;
 
-import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.opentripplanner.middleware.models.AdminUser;
+import org.opentripplanner.middleware.models.ApiKey;
+import org.opentripplanner.middleware.models.ApiUser;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.persistence.PersistenceUtil;
+import org.opentripplanner.middleware.utils.ApiGatewayUtils;
+import org.opentripplanner.middleware.utils.HttpUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.http.HttpResponse;
+
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static org.opentripplanner.middleware.OtpMiddlewareMain.getConfigPropertyAsText;
+import static org.opentripplanner.middleware.TestUtils.getBooleanEnvVar;
+import static org.opentripplanner.middleware.TestUtils.mockAuthenticatedRequest;
 
 /**
  * Tests for creating and deleting api keys. The following config parameters are required for these tests to run:
- *
- * DISABLE_AUTH set to true to bypass auth checks.
- * DEFAULT_USAGE_PLAN_ID set to a valid usage plan id. AWS requires this to create an api key.
- * OTP_MIDDLEWARE_URI_ROOT, e.g. http://localhost:4567.
+ *  - RUN_E2E=true the end-to-end environment variable must be set (NOTE: this is not a config value)
+ *  - An AWS_PROFILE is required, or AWS access has been configured for your operating environment e.g.
+ *    C:\Users\<username>\.aws\credentials in Windows or Mac OS equivalent.
+ *  - DISABLE_AUTH set to true to bypass auth checks and use users defined here. DEFAULT_USAGE_PLAN_ID set to a valid usage
```

Comment:
~Do `AWS_PROFILE` and `DEFAULT_USAGE_PLAN_ID` need to be added to `configurations/test/env.yml`?~
Although this concerns a test, the comment should state that because this is an E2E run, these variables need to be set in `configurations/default/env.yml`.